t/is-and-extract-ipv6.pl

```diff
@@ -0,0 +1,531 @@
+use Test::More;
+
+SKIP: { eval { require Test::Warnings; } or skip($@, 1); }
+eval { require 'ddclient'; } or BAIL_OUT($@);
+
+# Note, for GUA addresses we use 2001:DB8::/32 as that is reserved for
+# documentation per RFC 3849 and so not routable on public internet.
+# But we do have to add some others starting in 3xxx:: because they 
+# are also valid GUA.
+my @valid_ipv6_gua = (
+    "2001:DB8:4341:0781:1111:2222:3333:4444",
+    "2001:DB8:4341:0781::4444",
+    "2001:DB8:4341:0781:1111::",
+    "2001:DB8:4341:0781::100",
+    "2001:DB8:4341:0781::1",
+    "2001:DB8:4341:0781::0001",
+    "3001:DB8:4341:0781:1111:2222:3333:4444",
+    "3001:DB8:4341:0781::4444",
+    "3001:DB8:4341:0781:1111::",
+    "3001:DB8:4341:0781::100",
+    "3001:DB8:4341:0781::1",
+    "3001:DB8:4341:0781::0001",
+);
+
+# For ULA addresses we randomly generated a /48 prefix per RFC 4193
+# and are using subnet ID of 1 because why not.
+my @valid_ipv6_ula = (
+    "fdb6:1d86:d9bd:1:1111:2222:3333:4444",
+    "fdb6:1d86:d9bd:1::4444",
+    "fdb6:1d86:d9bd:1:1111::",
+    "fdb6:1d86:d9bd:1::100",
+    "fdb6:1d86:d9bd:1::1",
+    "fdb6:1d86:d9bd:1::0001",
+);
+
+# 
+my @valid_ipv6_lla = (
+    "fe80::1111:2222:3333:4444",
+    "fe80::4444",
+    "fe80::1111",
+);
```

Comment:
Include the first and last addresses of the block (fe80:: and febf:ffff:ffff:ffff:ffff:ffff:ffff:ffff) in this list.

Reply:
Added